.drone.yml

```diff
@@ -6,10 +6,25 @@ steps:
   - name: greeting
     image: alpine
     commands:
-      - echo hello
-      - echo world
+      - echo This step is to ensure that the pipeline itself is functioning properly.
+  - name: build
+    image: maven:3-jdk-11
+    commands:
+      - mvn -ntp -B clean
+      - mvn -ntp -B compile
   - name: test
     image: maven:3-jdk-11
     commands:
-      - mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V
-      - mvn test -B
+      - mvn -ntp install -DskipTests=true -Dmaven.javadoc.skip=true -B -V
```

Comment:
Why perform an install (skip the tests) and then test?

Reply:
See Maven's Lifecycles Reference: http://maven.apache.org/ref/3.6.3/maven-core/lifecycles.html

Basically there are phases that must be performed before other phases.  In this case test is always performed before install.

Reply:
Dunno, that's just the way that the drone tutorial did it.

Reply:
Looking more at the maven life cycles, it doesn't make sense to package an artifact without running unit test first.  Builds should be fast in order to provide constant feedback.  You want your build to fail as soon as something is wrong.  In this case unit tests.  We should just run the tests.  In the future we should have integration test but that should be another step.

Reply:
Let's remove the mvn install line and keep the mvn test line.